### PR TITLE
First pass at updating for tf12

### PR DESCRIPTION
**Pull request by cblkwell**
I went through and made the necessary changes to get this working in TF12; I was able to successfully create a lambda job in the sandbox account using this code.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -78,15 +78,15 @@ data "aws_iam_policy_document" "logs_policy_doc" {
...
 resource "aws_iam_role_policy" "main" {
   name = "lambda-${local.full_name}"
-  role = "${aws_iam_role.main.id}"
+  role = aws_iam_role.main.id
 
-  policy = "${data.aws_iam_policy_document.logs_policy_doc.json}"
+  policy = data.aws_iam_policy_document.logs_policy_doc.json
 }
 
 # This code verifies that the count of policy ARNs matches the actual
```
Comment (eeeady) [inline main.tf:92]: Do we still need this workaround? I'm not sure I understand the particular limitation that Terraform has/had here and if that got fixt in TF12?

**Review by eeeady — COMMENTED** [review]
Generally looks good to me. Had a question about that workaround local-exec thing though.

File: main.tf (hunk 1 above)
Comment (dynamike) [reply main.tf:92]: I've been thinking about potentially doing this in 2 phases 
1) to make our modules syntactically work with 0.12 
2) refactor the modules to more closely align with the 0.12 feature set.

That being said I'm open to other opinions.

Comment (cblkwell) [reply main.tf:92]: I *think* this was the Terraform issue that was forcing this workaround: https://github.com/hashicorp/terraform/issues/14677. This is supposedly fixed in 0.12 from the sound of it. I tend to agree with @dynamike that a two-stage process works pretty well. I also wanted to try and get this sorted out ASAP under the assumption we can make it prettier later.

Comment (eeeady) [reply main.tf:92]: in which case, I'll approve.

**Review by eeeady — APPROVED** [review]
YASSSSS UPGRADE

**Pull request merged**